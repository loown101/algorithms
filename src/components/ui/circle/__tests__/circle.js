import renderer from 'react-test-renderer';
import { ElementStates } from '../../../../types/element-states';
import { Circle } from '../circle';

describe('test Circle component', () => {
  const letterText = 'I am letter!';
  const isTrue = true;

  it('render Circle without letter', () => {
    const circle = renderer.create(<Circle />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with letter', () => {
    const circle = renderer.create(<Circle letter={letterText} />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with head', () => {
    const circle = renderer.create(<Circle head />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with element in head', () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with tail', () => {
    const circle = renderer.create(<Circle tail />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with element in tail', () => {
    const circle = renderer.create(<Circle tail={<Circle />} />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with index', () => {
    const circle = renderer.create(<Circle index />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with isSmall', () => {
    const circle = renderer.create(<Circle isSmall={isTrue} />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with state Default', () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />).toJSON()

    expect(circle).toMatchSnapshot()
  })

  it('render Circle with state Changing', () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />).toJSON()

    expect(circle).toMatchSnapshot()
  })


  it('render Circle with state Modified', () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />).toJSON()

    expect(circle).toMatchSnapshot()
  })

})