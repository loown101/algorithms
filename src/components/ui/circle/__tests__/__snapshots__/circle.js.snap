// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Circle component render Circle with element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`test Circle component render Circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    true
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`test Circle component render Circle with isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      I am letter!
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with state Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with state Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with state Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test Circle component render Circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
